RUFAS/routines/field/field/manure_application.py

```diff
@@ -724,18 +724,25 @@ def _determine_weighted_manure_attributes(
 
         """
         new_dry_matter_mass = old_total_dry_mass + application_dry_mass
-        application_moisture_factor = ManureApplication._determine_moisture_factor(application_dry_fraction)
-        new_moisture_factor = (
-            old_moisture_factor * old_total_dry_mass + application_moisture_factor * application_dry_mass
-        ) / new_dry_matter_mass
-        new_field_coverage = (
-            old_field_coverage * old_total_dry_mass + application_field_coverage * application_dry_mass
-        ) / new_dry_matter_mass
-        return {
-            "new_dry_matter_mass": new_dry_matter_mass,
-            "new_moisture_factor": new_moisture_factor,
-            "new_field_coverage": new_field_coverage,
-        }
+        if new_dry_matter_mass > 0:
+            application_moisture_factor = ManureApplication._determine_moisture_factor(application_dry_fraction)
+            new_moisture_factor = (
+                old_moisture_factor * old_total_dry_mass + application_moisture_factor * application_dry_mass
+            ) / new_dry_matter_mass
+            new_field_coverage = (
+                old_field_coverage * old_total_dry_mass + application_field_coverage * application_dry_mass
+            ) / new_dry_matter_mass
+            return {
+                "new_dry_matter_mass": new_dry_matter_mass,
+                "new_moisture_factor": new_moisture_factor,
+                "new_field_coverage": new_field_coverage,
+            }
+        else:
+            return {
+                "new_dry_matter_mass": 0,
+                "new_moisture_factor": 0,
+                "new_field_coverage": 0,
+            }
```

Comment:
```suggestion
        if new_dry_matter_mass == 0.0:
            return {
                "new_dry_matter_mass": 0.0,
                "new_moisture_factor": 0.0,
                "new_field_coverage": 0.0,
            }
        application_moisture_factor = ManureApplication._determine_moisture_factor(application_dry_fraction)
        new_moisture_factor = (
            old_moisture_factor * old_total_dry_mass + application_moisture_factor * application_dry_mass
        ) / new_dry_matter_mass
        new_field_coverage = (
            old_field_coverage * old_total_dry_mass + application_field_coverage * application_dry_mass
        ) / new_dry_matter_mass
        return {
            "new_dry_matter_mass": new_dry_matter_mass,
            "new_moisture_factor": new_moisture_factor,
            "new_field_coverage": new_field_coverage,
        }
```
I suggest trying to add as little indentation as possible, but looks good either way.